psd.h

```diff
@@ -285,22 +496,30 @@ namespace psd {
     virtual void onLayersAndMask(const LayersInformation& layers) { }
     virtual void onLayersInfo(const LayersInformation& layers) { }
     virtual void onImageData(const ImageData& imageData) { }
+    virtual void onLayerSelected(const LayerRecord &layer){}
+    virtual void onLayerDeselected(const LayerRecord &layer){}
```

Comment:
Follow the existing style and we can use some names like in onBeginImage:
```suggestion
    virtual void onBeginLayer(const LayerRecord &layer) { }
    virtual void onEndLayer(const LayerRecord &layer) { }
```